packages/cli/src/ui/components/shared/MaxSizedBox.tsx

```diff
@@ -24,6 +24,30 @@ export function setMaxSizedBoxDebugging(value: boolean) {
   enableDebugLog = value;
 }
 
+export const KEYWORDS_TEXT_INK = [
+  'constructor',
+  'toString',
+  'valueOf',
+  'hasOwnProperty',
+  'isPrototypeOf',
+  'propertyIsEnumerable',
+  'toLocaleString',
+  '__proto__',
+];
+
+/* 
+  When <Text>'string'</Text> is stored in const visibleLines and is printed inside a <Box>.
+  The string becomes a children prop of Text node. This causes issues with printing keywords. 
+  This function appends a zero-width space(\u200B) to the keyword so that it escapes misinterpretation.
+  */
+function handleKeywordsForTextComponent(text: string) {
+  if (KEYWORDS_TEXT_INK.includes(text)) {
```

Comment:
this is really surprising. We should test if we can reproduce the issue with a simple test case using just the Ink library. If so, likely there is a case where we need to fix how ink is doing property access. For example, likely there is a case where Ink needs to use `hasOwnProperty` or similar rather than using an unsafe method like `['propertyName']`